src/emotion.d.ts

```diff
@@ -0,0 +1,9 @@
+import '@emotion/react';
+import theme from './theme';
+
+type ThemeType = typeof theme;
+
+declare module '@emotion/react' {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  export interface Theme extends ThemeType {}
```

Comment:
See https://emotion.sh/docs/typescript#define-a-theme; this is how to type emotion's `Theme`.

And we seem to basically get our theme shape for free by simply `extend`ing the `typeof` our `theme`.